api/python/provisioner/commands/setup_timeserver.py

```diff
@@ -0,0 +1,52 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+"""Defines a module to setup time server on system"""
+
+import logging
+from typing import Type
+
+from . import RunArgsEmpty
+from ._basic import CommandParserFillerMixin
+from ..errors import SaltError
+from ..vendor import attr
+from ..config import LOCAL_MINION
+from ..pillar import (
+    PillarKey,
+    PillarResolver
+)
+from ..salt import function_run, StatesApplier
+from .. import inputs, values
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class SetupTimeServer(CommandParserFillerMixin):
+    """Defines logic to setup time server on system"""
+
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = RunArgsEmpty
+
+    def run(self):
+        logger.info("Setting up NTP time server")
+        try:
+            StatesApplier.apply(['components.system.chrony'], targets=LOCAL_MINION)
```

Comment:
we have similar logic as `set_ntp` command and it allows configuration during the runtime (on field). On any change of configuration it calls [config, restart for noes and config for controller](https://jts.seagate.com/secure/XrayTestRepositoryAction!default.jspa?entityKey=TEST&path=EOS%2FEES%2FProvisioner#). 

I guess that new API is considered for deployment only? Is it true? What about controllers configuration?

Reply:
@andkononykhin2 This api is for `Field: Node config` and here we are not considering time_server details at command-line. 
also currently the requirement does not include NTP configuration on controller 

hence we decided to keep both the logic's different.

Reply:
I tend to resolve that discussion for now. Just few general thoughts:

- it's not a good UX if we have multiple API for the similar things: that configuration anyway requires settings in pillar the difference is:
  -  in one case we consider a change to pillar
  - in another - we don't consider change but just want to apply/re-apply

I don't think we should publish these APIs for now. I believe we need more discussion on that and implement just one API that satisfies both cases. (Actually I think `set_ntp` already does that: you can pass special value `PRVSNR_UNCHANGED` so it won't update the pillar but just re-apply).

Reply:
CC @ypise

Reply:
@andkononykhin2  removed setup_timeserver.py so that we can reuse `set_ntp`